From 286e3db76c2590ba6aa14ac73025479b622ee314 Mon Sep 17 00:00:00 2001
From: Quallenauge <Hamsi2k@freenet.de>
Date: Sat, 19 Feb 2022 22:28:20 +0100
Subject: [PATCH 2/3] Biometrics: Allow disabling of fingerprint cleanups

Don't schedule cleanups at all if the driver
doesn't support enumerate function.

Change-Id: If9e1b82bc551e2fd06218b1720f5986633c1ab55
---
 .../sensors/fingerprint/aidl/FingerprintProvider.java     | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/services/core/java/com/android/server/biometrics/sensors/fingerprint/aidl/FingerprintProvider.java b/services/core/java/com/android/server/biometrics/sensors/fingerprint/aidl/FingerprintProvider.java
index c18d92530135..ac05e93bcefa 100644
--- a/services/core/java/com/android/server/biometrics/sensors/fingerprint/aidl/FingerprintProvider.java
+++ b/services/core/java/com/android/server/biometrics/sensors/fingerprint/aidl/FingerprintProvider.java
@@ -137,6 +137,8 @@ public class FingerprintProvider implements IBinder.DeathRecipient, ServiceProvi
     @Nullable private IVirtualHal mVhal;
     @Nullable private String mHalInstanceNameCurrent;
 
+    private boolean mCleanup;
+
     private final class BiometricTaskStackListener extends TaskStackListener {
         @Override
         public void onTaskStackChanged() {
@@ -206,6 +208,9 @@ public class FingerprintProvider implements IBinder.DeathRecipient, ServiceProvi
         mTestHalEnabled = testHalEnabled;
         mBiometricHandlerProvider = biometricHandlerProvider;
 
+        mCleanup = context.getResources().getBoolean(
+                org.lineageos.platform.internal.R.bool.config_cleanupUnusedFingerprints);
+
         initAuthenticationBroadcastReceiver();
         initFingerprintDanglingBroadcastReceiver();
         initSensors(resetLockoutRequiresHardwareAuthToken, props, gestureAvailabilityDispatcher);
@@ -674,6 +679,9 @@ public class FingerprintProvider implements IBinder.DeathRecipient, ServiceProvi
     @Override
     public void scheduleInternalCleanup(int sensorId, int userId,
             @Nullable ClientMonitorCallback callback, boolean favorHalEnrollments) {
+        if (!mCleanup) {
+            return;
+        }
         mHandler.post(() -> {
             final FingerprintInternalCleanupClient client =
                     new FingerprintInternalCleanupClient(mContext,
-- 
2.39.5

